lib/document.js

```diff
@@ -3950,7 +3957,7 @@ Document.prototype.toJSON = function(options) {
  */
 
 Document.prototype.parent = function() {
-  return this.$__.parent;
+  return this.$__.parent || this;
```

Comment:
This change is incorrect: a subdocument can have a null parent. At the very least, you can do `if (this.$__.isSubdocument) { return this.$__.parent; } return this;`